shared-module/is31fl3741/is31fl3741.h

```diff
@@ -0,0 +1,161 @@
+/*
+ * This file is part of the Micro Python project, http://micropython.org/
+ *
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2021 Mark Komus
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+#pragma once
+
+#include "py/obj.h"
+#include "lib/protomatter/src/core.h"
+#include "shared-bindings/busio/I2C.h"
+
+extern const mp_obj_type_t is31fl3741_is31fl3741_type;
+typedef struct {
+    mp_obj_base_t base;
+    mp_obj_t framebuffer;
+    mp_buffer_info_t bufinfo;
+    uint16_t bufsize, width, height, scale_width, scale_height;
+    busio_i2c_obj_t *i2c;
+    busio_i2c_obj_t inline_i2c;
+    uint8_t device_address;
+    uint8_t bit_depth;
+    bool paused;
+    bool scale;
+    bool auto_gamma;
+} is31fl3741_is31fl3741_obj_t;
+
+static const uint16_t glassesmatrix_ledmap[18 * 5 * 3] = {
```

Comment:
Looking at ideas on how to pass this data into the display driver

Reply:
Passing in a byte buffer is how I'd do it. I'd assume we'd provide libraries that do the init. So, most folks wouldn't need to worry about it.

Reply:
Remove this please since it is passed in now.